File: CS/DevExpress.Calendar.Model/Resource.cs
namespace DevExpress.Calendar.Model {
    using System;
    using System.Collections.Generic;
    using System.Text;
    using DevExpress.Xpo;
    using DevExpress.Xpo.Metadata;
#if !WIN
    using DevExpress.Xpo.Services;
    using DevExpress.Data.Services;
#endif

#if !WIN
    [ResourceSetName("Resources")]
#endif
    [Persistent("Resources")]
    public partial class Resource : XPCustomObject {
        public Resource(Session session)
            : base(session) {
        }

        public Resource() {            
        }

        public Resource(Session session, XPClassInfo classInfo)
            : base(session, classInfo) {
        }         

        Guid _id;
        [Key(AutoGenerate = true)]
        public Guid ID {
            get {
                return _id;
            }
            set {
                SetPropertyValue("ID", ref _id, value);
            }
        }

        string _email;
        [Size(256)]
        public string Email {
            get {
                return _email;
            }
            set {
                SetPropertyValue("Email", ref _email, value);
            }
        }

        string _userName;
        [Size(256)]
        public string UserName {
            get {
                return _userName;
            }
            set {
                SetPropertyValue("UserName", ref _userName, value);
            }
        }

        string _name;
        [Size(256)]
        public string Name {
            get {
                return _name;
            }
            set {
                SetPropertyValue("Name", ref _name, value);
            }
        }

        [Association("Appointments-Resources", UseAssociationNameAsIntermediateTableName = true)]
        public XPCollection<Appointment> Appointments {
            get {
                return GetCollection<Appointment>("Appointments");
            }
        }
    }
}

File: VB/DevExpress.Calendar.Model/Resource.vb
Imports Microsoft.VisualBasic
	Imports System
	Imports System.Collections.Generic
	Imports System.Text
	Imports DevExpress.Xpo
	Imports DevExpress.Xpo.Metadata
	Imports DevExpress.Xpo.Services
	Imports DevExpress.Data.Services
Namespace DevExpress.Calendar.Model
#If (Not WIN) Then
#End If

#If (Not WIN) Then
    <ResourceSetName("Resources")> _
Partial Public Class Resource
        Inherits XPCustomObject
#Else
<Persistent("Resources")> _
	Partial Public Class Resource
		Inherits XPCustomObject
#End If

        Public Sub New(ByVal session As Session)
            MyBase.New(session)
        End Sub

        Public Sub New()
        End Sub

        Public Sub New(ByVal session As Session, ByVal classInfo As XPClassInfo)
            MyBase.New(session, classInfo)
        End Sub

        Private _id As Guid
        <Key(AutoGenerate:=True)> _
        Public Property ID() As Guid
            Get
                Return _id
            End Get
            Set(ByVal value As Guid)
                SetPropertyValue("ID", _id, value)
            End Set
        End Property

        Private _email As String
        <Size(256)> _
        Public Property Email() As String
            Get
                Return _email
            End Get
            Set(ByVal value As String)
                SetPropertyValue("Email", _email, value)
            End Set
        End Property

        Private _userName As String
        <Size(256)> _
        Public Property UserName() As String
            Get
                Return _userName
            End Get
            Set(ByVal value As String)
                SetPropertyValue("UserName", _userName, value)
            End Set
        End Property

        Private _name As String
        <Size(256)> _
        Public Property Name() As String
            Get
                Return _name
            End Get
            Set(ByVal value As String)
                SetPropertyValue("Name", _name, value)
            End Set
        End Property

        <Association("Appointments-Resources", UseAssociationNameAsIntermediateTableName:=True)> _
        Public ReadOnly Property Appointments() As XPCollection(Of Appointment)
            Get
                Return GetCollection(Of Appointment)("Appointments")
            End Get
        End Property
    End Class
End Namespace